### ISSUE TITLE
rgdp compliance

### ISSUE THREAD

**Issue by pmaojo** [infinum/cookies_eu#60]
This plugin is ok with the new law?

**Comment by damonbakker** [comment]
I think there needs to be some form of modification.
The user should be able to disable all optional categories of cookies. Only neccessary cookies shouldn't be able to be opted out of by the user as far as I'm aware.

**Comment by pmaojo** [comment]
Own cookies (session for example) are needed and should not be blocked only third party cookies are a problem for the law.

**Comment by seanfcarroll** [comment]
Do you mean GDPR compliance?

**Comment by gabskoro** [comment]
@seanfcarroll I think he just made a typo :)

@pmaojo So, how did you imagine that? Having a "Not agree" button which will call a method in which you can do whatever you want?

**Comment by pmaojo** [comment]
GDPR in English
RGPD in Spanish

**Comment by pmaojo** [comment]
@gabskoro The button should be "Agree", but if the Agree cookie is not set, all cookies should be rejected. In Spain, in theory, third party cookies that store data that could identify user should be disabled until user allows them.

**Comment by hlascelles** [comment]
Yes, looks like this is still a necessary change. We probably also need a configurable list of cookies, put into categories (mandatory, analytics etc...), that the user can opt into and out of.

**Comment by gabskoro** [comment]
I think this gem should not be in charge of that. Maybe we could just render a "NO" button which will trigger a specific route and then you can block whatever you would like. What do you think?
